### DISCUSSION TITLE
Nx affected dynamic base branch

### DISCUSSION THREAD

**Discussion by sandeb-420** [nrwl/nx#13168] [General]
Hi,

I am trying to integrate nx workspace for an angular multi-project setup.
I did get most of the setup right and successfully leveraging NX’s capabilities apart from one thing.

Basically I want to run tests prepush using git hooks and want to use the affected command. What I don’t understand is how can I make the defaultbase dynamic?

For patch/bugfixes I may be cutting branches from different release branch. So do I need to manually change the base branch each time?
My understanding is NX uses this as base branch to calculate diff from HEAD. Or maybe I am missing something here?

Any help is much appreciated

**Comment by veloware** [comment]
you can specify a different `base` or `head` in cli

e.g.

```
nx affected --target=test --base=<NEW_BASE> --head=<NEW_HEAD>
```

https://nx.dev/nx/affected
